src/layer/cursor.js

```diff
@@ -211,25 +210,27 @@ var Cursor = function(parentEl) {
         // cache for textarea and gutter highlight
         this.$pixelPos = pixelPos;
         this.restartTimer();
-    };
-    
-    this.drawCursor = null;
+    }
 
-    this.$setOverwrite = function(overwrite) {
+    $setOverwrite(overwrite) {
         if (overwrite != this.overwrite) {
             this.overwrite = overwrite;
             if (overwrite)
                 dom.addCssClass(this.element, "ace_overwrite-cursors");
             else
                 dom.removeCssClass(this.element, "ace_overwrite-cursors");
         }
-    };
+    }
 
-    this.destroy = function() {
+    destroy() {
         clearInterval(this.intervalId);
         clearTimeout(this.timeoutId);
-    };
+    }
+
+}
+
+Cursor.prototype.$padding = 0;
+Cursor.prototype.drawCursor = null;
```

Comment:
Any reason why we still define some properties via the prototype instead of defining them as class properties? If it's because they're static, can we define them as [`static` class properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static)?

Left a similar comment above; there are more examples like this too.